### PR TITLE
DPRINT/Watcher support for idle ethernet cores

### PR DESCRIPTION
**Pull request by tt-dma**
CI running: https://github.com/tenstorrent/tt-metal/actions/runs/8806874607

### CODE REVIEW

File: tt_metal/hw/inc/debug/pause.h [hunk 1]
```diff
@@ -5,13 +5,18 @@
...
 
 #if defined(WATCHER_ENABLED) && !defined(WATCHER_DISABLE_PAUSE)
 
 void watcher_pause() {
     // Write the pause flag for this core into the memory mailbox for host to read.
     debug_pause_msg_t tt_l1_ptr *pause_msg = GET_MAILBOX_ADDRESS_DEV(pause_status);
     pause_msg->flags[debug_get_which_riscv()] = 1;
+    WATCHER_RING_BUFFER_PUSH((uint32_t) pause_msg);
+    WATCHER_RING_BUFFER_PUSH(debug_get_which_riscv());
+    DEBUG_STATUS('P', 'A', 'U', 'S');
```
Comment (pgkeller) [inline tt_metal/hw/inc/debug/pause.h:19]: let's make this PASW and add a PASD after

File: tt_metal/impl/device/device.hpp [hunk 2]
```diff
@@ -118,10 +118,20 @@ class Device {
         return tt::Cluster::instance().get_active_ethernet_cores(this->id_, skip_reserved_tunnel_cores);
     }
 
+    bool is_active_ethernet_core(CoreCoord logical_core, bool skip_reserved_tunnel_cores=false) const {
+        return tt::Cluster::instance().get_active_ethernet_cores(this->id_, skip_reserved_tunnel_cores).find(logical_core)
```
Comment (pgkeller) [inline tt_metal/impl/device/device.hpp:122]: it's not clear to me that the compiler will optimize away the 2nd lookup, grab the value and use it twice:
tmp = tt::Cluster::instance().get_active_ethernet_cores(this->id_, skip_reserved_tunnel_cores);
tmp.find() != tmp.last();

**Review by pgkeller — CHANGES_REQUESTED** [review]

File: tt_metal/hw/inc/debug/pause.h (hunk 1 above)
Comment (tt-dma) [reply tt_metal/hw/inc/debug/pause.h:19]: Oops was using this for debugging, will update it to a better value

File: tt_metal/impl/device/device.hpp (hunk 2 above)
Comment (tt-dma) [reply tt_metal/impl/device/device.hpp]: Fixed

**Comment by tt-dma** [comment]
CI passing after extra changes/rebase: https://github.com/tenstorrent/tt-metal/actions/runs/8808751748

**Review by pgkeller — APPROVED** [review]

**Pull request merged**